### ISSUE TITLE
Core Dumped

### ISSUE THREAD

**Issue by Edomfra** [marbl/canu#2177]
**Core Dumped**

I did try use canu for assembly a 18s gene with a size aprox 1.7K.

**command**:`../canu/build/bin/canu -p prs_meloidogyne readSamplingCoverage=100 genomeSize=2k  -nanopore ../wordir/sample1.fastq `
 `canu -version: canu snapshot v2.3-development +79 changes (r10350 48786ab3beecb04269a29934de96d06cd931a802)` 

 **system**: Linux (ubuntu)
 **size of file**: 121Mb 

```
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:

ABORT:
ABORT: canu snapshot v2.3-development +79 changes (r10350 48786ab3beecb04269a29934de96d06cd931a802)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to partition reads for correction.
ABORT:
ABORT: Disk space available:  7243.787 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (correction/2-correction/correctReadsPartition.err):
ABORT:
ABORT:   
ABORT:   Found perl:
ABORT:      /opt/bin/anaconda3/bin/perl
ABORT:      This is perl 5, version 22, subversion 0 (v5.22.0) built for x86_64-linux-thread-multi
ABORT:   
ABORT:   Found java:
ABORT:      /opt/bin/anaconda3/bin/java
ABORT:      openjdk version "11.0.1" 2018-10-16 LTS
ABORT:   
ABORT:   Found canu:
ABORT:      /opt/bin/canu/build/bin/canu
ABORT:      canu snapshot v2.3-development +79 changes (r10350 48786ab3beecb04269a29934de96d06cd931a802)
ABORT:   
ABORT:   -- Opening seqStore '../../prs_meloidogyne.seqStore'.
ABORT:   -- Opening corStore '../prs_meloidogyne.corStore' version 1.
ABORT:   -- Loading list of reads to process from './prs_meloidogyne.readsToCorrect'.
ABORT:   free(): invalid pointer
ABORT:   
ABORT:   Failed with 'Aborted'; backtrace (libbacktrace):
ABORT:   utility/src/utility/system-stackTrace.C::82 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   stores/sqCache.C::80 in _ZN7sqCacheD2Ev()
ABORT:   correction/falconsense.C::732 in main()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   Aborted (core dumped)
ABORT:
```

**Comment by skoren** [comment]
I'd guess you're ending up with no reads to correct. Post the full log from the run. Have you tried using the pre-compiled binary to make sure it's not the build?

**Comment by Edomfra** [comment]
I did try with a version precompiled from conda  and and other using git clone, using the conda version I had trouble with the size of the reads. 

These is the file correctReadsPartition.err output from the version (git clone) `canu snapshot v2.3-development +79 changes (r10350 48786ab3beecb04269a29934de96d06cd931a802)`  
```
Found perl:
   /opt/bin/anaconda3/bin/perl
   This is perl 5, version 22, subversion 0 (v5.22.0) built for x86_64-linux-thread-multi

Found java:
   /opt/bin/anaconda3/bin/java
   openjdk version "11.0.1" 2018-10-16 LTS

Found canu:
   /opt/bin/canu/build/bin/canu
   canu snapshot v2.3-development +79 changes (r10350 48786ab3beecb04269a29934de96d06cd931a802)

-- Opening seqStore '../../prs_meloidogyne.seqStore'.
-- Opening corStore '../prs_meloidogyne.corStore' version 1.
-- Loading list of reads to process from './prs_meloidogyne.readsToCorrect'.
free(): invalid pointer

Failed with 'Aborted'; backtrace (libbacktrace):
utility/src/utility/system-stackTrace.C::82 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
stores/sqCache.C::80 in _ZN7sqCacheD2Ev()
correction/falconsense.C::732 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
Aborted (core dumped)

```
the file *.readsToCorrect.err only print `Bye.`

the output from *.readsToCorrect.stats
```

                          original      original
                         raw reads     raw reads
category                w/overlaps  w/o/overlaps
-------------------- ------------- -------------
Number of Reads                511          1189
Number of Bases             777030         10629
Coverage                    97.129         1.329
Median                        1389             0
Mean                          1520             8
N50                           1520          3008
Minimum                       1004             0
Maximum                       3651          3008

                                     --------corrected---------  ----------rescued----------
                          evidence                     expected                     expected
category                     reads            raw     corrected            raw     corrected
-------------------- -------------  ------------- -------------  ------------- -------------
Number of Reads                495            167           167              9             9
Number of Bases             751933         332532        320325          10894         10162
Coverage                    93.992         41.566        40.041          1.362         1.270
Median                        1400           1894          1867           1209          1160
Mean                          1519           1991          1918           1210          1129
N50                           1539           1911          1888           1217          1164
Minimum                       1000           1548          1535           1155          1022
Maximum                       3651           3651          2467           1293          1228

                     --------uncorrected--------
                                        expected
category                       raw     corrected
-------------------- ------------- -------------
Number of Reads               1524          1524
Number of Bases             444233        340476
Coverage                    55.529        42.559
Median                           0             0
Mean                           291           223
N50                           1265          1171
Minimum                          0             0
Maximum                       3547          3444

Maximum Memory           558299306

```

**Comment by skoren** [comment]
What do you mean trouble with the size of the reads? I think the error you're hitting is a build problem on your system and I'd definitely suggest downloading the pre-compiled binary for your system from GitHub (https://github.com/marbl/canu/releases, you don't need conda) and running that instead.

**Comment by brianwalenz** [comment]
Hopefully fixed in b415de9b171cc4e1a2aeba6a2da91d5b12595518.

**Comment by Edomfra** [comment]
Thanks , I tried with pre-compiled version 2.2 and run succsessfully
